src/it/scala/org/apache/spark/sql/snowflake/SFQueryTest.scala

```diff
@@ -62,6 +62,19 @@ object SFQueryTest {
     case (a: java.math.BigDecimal, b: Float) =>
       (a.toString.toFloat - b).abs < 0.0001
     case (a: java.math.BigDecimal, b: java.math.BigDecimal) => a.equals(b)
```

Comment:
to reviewers:
`scala.math.BigDecimal` and `java.math.BigDecimal` are two different classes.
Some testing data is using `scala.math.BigDecimal`, this function needs to be updated to handle it like `java.math.BigDecimal`.

Reply:
Why are all of these additions needed?